transforms/geocable.py

```diff
@@ -645,19 +672,99 @@ def interpolate_rope(node1: Node, node2: Node, seg_count: int) -> List[Node]:
         for point in moveable
     ]
 
+def compute_binominal(n, k):
+    # n and k must be integers
+    value = 1.0
+    for i in range(1,k+1):
+        value = value * ((n + 1 - i) / i)
+    if n == k:
+        value = 1
+    return value
+
+
+def interpolate_bezier_quad(first_node:Node,last_node:Node,curve_segment_count:int) -> List[Node]:
+    """Interpolate a bezier curve, for better 90 degrees turn."""
+    # reference:
+    # https://en.wikipedia.org/wiki/De_Casteljau%27s_algorithm
+    points: list[Node] = []
+    increment = 1/(curve_segment_count)
+    # Only the segment count set in the first spline object counts
+    curve_x = []
+    curve_y = []
+    curve_z = []
+    curnode = first_node
+    while curnode != None:
+        curve_x.append(curnode.pos.x)
+        curve_y.append(curnode.pos.y)
+        curve_z.append(curnode.pos.z)
+        curnode = curnode.next
+    # We do not calculate the first and last node of this curve
+    for l in range(1,curve_segment_count-1):
+        t = l * increment
+        x = de_casteljau(t,curve_x)
+        y = de_casteljau(t,curve_y)
+        z = de_casteljau(t,curve_z)
+        n = Node(Vec(x,y,z), first_node.config, first_node.radius)
+        points.append(n)
+    LOGGER.debug("Bezier Nodes Generated: ",points)
+    return points
+
+def de_casteljau(t, coefs):
+    beta = [c for c in coefs] # values in this list are overridden
+    n = len(beta)
+    for j in range(1, n):
+        for k in range(n - j):
+            beta[k] = beta[k] * (1 - t) + beta[k + 1] * t
+    return beta[0]
+
+def find_all_connected_exclude_firstlast(node: Node):
+    node_list : List[Node] = [node]
+    cur_back = node
+    cur_forward = node
+
+    while cur_back.prev != None:
+        cur_back = cur_back.prev
+        node_list.append(cur_back)
+        assert cur_back.prev != node, 'Circular Node Detected'
```

Comment:
It is possible to create a loop in nodes, have you tested what happens? (I don't remember if I've made it break up such node sets.)

Reply:
With the original implementation the postcompiler goes into infinite loop.